src/main/java/hudson/tasks/test/AbstractTestResultAction.java

```diff
@@ -332,11 +587,24 @@ private Area calcDefaultSize() {
         else
             return new Area(500,200);
     }
-    
+
+    /**
+     * A method to build the dataset to be used for generating trends.
+     * @param req The HTTP request message for the trend.
+     * @return An object of type {@link CategoryDataset} in which columns are the build numbers to be depicted on
+     * x-axis and rows are the different data series that need to be analysed.
+     * <p>
+     * This method creates {@link CategoryDataset} object and the created object is exactly same as the one
+     * created by {@link #buildDataSetPerProject(StaplerRequest)} with the only difference being it does not
+     * work for a particular project but only for "AllProjects" option.
+     * <p>
+     * This method is retained though its functionality is subset of functionality of
+     * {@link #buildDataSetPerProject(StaplerRequest)} as it was there in older versions also.
+     */
     private CategoryDataset buildDataSet(StaplerRequest req) {
-        boolean failureOnly = Boolean.valueOf(req.getParameter("failureOnly"));
+        boolean failureOnly = Boolean.valueOf(getParameter(req,FAILUREONLY));
 
-        DataSetBuilder<String,NumberOnlyBuildLabel> dsb = new DataSetBuilder<String,NumberOnlyBuildLabel>();
+        DataSetBuilder<String,NumberOnlyBuildLabel> dsb = new DataSetBuilder<>();
```

Comment:
why this change ?

Reply:
Did undo regarding this change. So, this change is not there anymore.